### PR TITLE
Make DockerSaveImage out of date if the input state on the daemon changed (#1071)

### LINKED ISSUE #1071: DockerSaveImage should probably never be up-to-date
`DockerSaveImage` depends on the state of the Docker daemon as input.
It should probably never be up-to-date, and thus annotated with `@UntrackedTask`.
I have a `DockerBuildImage` task that builds the docker image from the Docker file and then a `DockerSaveTask` that saves it to a file that I then release as artifact.
But as long as none of the inputs or outputs of the task changed, it is considered up-to-date even if the state on the Docker Daemon changed due to a rebuild.

### PR DESCRIPTION
**Pull request by Vampire**
Fixes #1071
Followup of #1067, hence still draft, otherwise top commit could already be reviewed

### CODE REVIEW

**Comment by bmuschko** [comment]
I merged the other PR. Can you rebase before I have a look at this one?

**Comment by Vampire** [comment]
Definitely, that's why it still was draft. :-)
Read for review now.

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerSaveImage.groovy [hunk 1]
```diff
@@ -35,14 +37,74 @@ class DockerSaveImage extends AbstractDockerRemoteApiTask {
...
     DockerSaveImage() {
         useCompression.set(false)
+        String safeTaskPath = path.replaceFirst("^:", "").replaceAll(":", "_")
+        imageIdsFile.set(project.layout.buildDirectory.file(".docker/${safeTaskPath}-imageIds.properties"))
+
         onlyIf {
             images.getOrNull()
         }
+
+        outputs.upToDateWhen {
```
Comment (bmuschko) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerSaveImage.groovy:64]: This method became quite long. Could you break this up a little into other methods? After that I think we are ready to merge.

Comment (Vampire) [reply src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerSaveImage.groovy:64]: Yeah, sure.
I stupidly had the code twice anyway for reading and writing.
Hope it suits better now. :-)

**Comment by bmuschko** [comment]
Thanks, this has been merged.

**Pull request merged**